core/engine.cpp

```diff
@@ -36,6 +36,26 @@
 #include "core/version.h"
 #include "core/version_hash.gen.h"
 
+// Instead of directly setting the frame time from the value from the corresponding OS API timing function
+// we now process it to deal with OS bugs that can cause it to move BACKWARDS in time.
+uint64_t Engine::update_idle_frame_ticks(uint64_t os_time)
+{
+	// prevent clock going backwards due to OS timer bugs
+	if (os_time > _frame_ticks)
+	{
```

Comment:
Style issue. See http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

Reply:
Ah yep sure I'll fix these with the beautifier plugin, it is just WIP for now as it needs thorough testing. :+1: